lib/astarte_vmq_plugin/queries.ex

```diff
@@ -130,8 +132,11 @@ defmodule Astarte.VMQ.Plugin.Queries do
   end
 
   defp use_realm(conn, realm) when is_binary(realm) do
-    with :ok <- verify_realm(realm),
-         {:ok, %Xandra.SetKeyspace{}} <- Xandra.execute(conn, "USE #{realm}") do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id())
+
+    with :ok <- verify_realm(keyspace_name),
```

Comment:
I think that `verify_realm` should be done for the realm, rather than the keyspace name (which is valid by construction as the output of `CQLUtils.realm_name_to_keyspace_name`)